internal/variants/variants.go

```diff
@@ -71,8 +81,29 @@ func (q *Query) ValidateInput() error {
 	if q.Allele == "" {
 		return errors.New("missing allele")
 	}
-	if q.Coord == nil {
-		return errors.New("missing coordinate")
+	if err := q.validateCoordinates(); err != nil {
+		return fmt.Errorf("validating coordinates: %v", err)
+	}
+	return nil
+}
+
+func (q *Query) validateCoordinates() error {
+	var precisePosition, imprecisePosition bool
+	if q.Start != nil && (q.End != nil || q.Allele != "") {
+		precisePosition = true
+	}
+	if q.StartMin != nil && q.StartMax != nil && q.EndMin != nil && q.EndMax != nil {
+		imprecisePosition = true
```

Comment:
What about the case where `q.StartMin = q.StartMax` and `q.EndMin = q.EndMax`?  Isn't that actually a precise position?

@anamanolache This is not a bad start, but please try to review the Beacon specification again and take some time to explore the scenarios that might occur.  The simplifications will become obvious, I promise you :)